File: src/main/java/parser/ast/expressions/VariableExpression.java
package parser.ast.expressions;

import stdlib.Value;
import stdlib.Variables;

public final class VariableExpression implements Expression {

    private final String name;

    public VariableExpression(String name) {
        this.name = name;
    }

    @Override
    public Value eval() {
        if (Variables.isNotExists(name)){
            throw new RuntimeException("Constant does not exists!");
        }

        return Variables.get(name);
    }

    @Override
    public String toString() {
        return String.format("<VariableExpression name=\"%s\"/>", name);
    }
}

File: src/main/java/parser/ast/expressions/VariableExpression.kt
package parser.ast.expressions

import stdlib.Value
import stdlib.Variables

class VariableExpression(private val name: String) : Expression {

    override fun eval(): Value {
        if (Variables.isNotExists(name)) {
            throw RuntimeException("Constant does not exists!")
        }

        return Variables[name]!!
    }

    override fun toString(): String {
        return String.format("<VariableExpression name=\"%s\"/>", name)
    }
}
